src/Interpreters/ActionsDAG.cpp

```diff
@@ -531,6 +561,115 @@ bool ActionsDAG::empty() const
     return true;
 }
 
+ActionsDAGPtr ActionsDAG::makeConvertingActions(
+    const ColumnsWithTypeAndName & source,
+    const ColumnsWithTypeAndName & result,
```

Comment:
What is result?

Reply:
Result is `ActionDAG` which converts block structure from source to result

Reply:
`result` is expected structure we should have after applying ActionsDAG